### PR TITLE
Adds LIME textexplainer, NER, POS, heatmaps, ROC and PR curves using wandb.log()

### PR DESCRIPTION
**Pull request by lavanyashukla**
Lets users log heatmaps, ROC and PR curves using `wandb.log()` like so:

```
# Visualize model performance
wandb.log({'roc': wandb.plots.ROC(y_test, y_probas, labels)})
wandb.log({'pr': wandb.plots.precision_recall(y_test, y_probas, labels)})

```

### CODE REVIEW

File: wandb/plots/heatmap.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
+import wandb
+from wandb import util
+from wandb.plots.utils import test_missing, test_types, encode_labels
+np = util.get_module("numpy", required="Logging plots requires numpy")
+scikit = util.get_module("sklearn")
```
Comment (vanpelt) [inline wandb/plots/heatmap.py]: We should put these in the heatmap method itself.   Putting a required get_module here will make it required for the whole library because we import `wandb.plots` in the `wandb` module.

File: wandb/plots/plot_definitions.py [hunk 2]
```diff
@@ -716,3 +716,53 @@
   ]
 }
 '''
+
```
Comment (vanpelt) [inline wandb/plots/plot_definitions.py]: What is this file for?  If this is just for reference let's turn it into a markdown file.

File: wandb/plots/precision_recall.py [hunk 3]
```diff
@@ -0,0 +1,92 @@
+import wandb
+from wandb import util
+from wandb.plots.utils import test_missing, test_types, encode_labels
+np = util.get_module("numpy", required="Logging plots requires numpy")
+scikit = util.get_module("sklearn")
+chart_limit = wandb.Table.MAX_ROWS
```
Comment (vanpelt) [inline wandb/plots/precision_recall.py]: Optional deps should go in the func.  Also, if scikit is required we should make it so.

File: wandb/plots/roc.py [hunk 4]
```diff
@@ -0,0 +1,69 @@
+import wandb
+from wandb import util
+from wandb.plots.utils import test_missing, test_types, encode_labels
+np = util.get_module("numpy", required="Logging plots requires numpy")
+scikit = util.get_module("sklearn")
```
Comment (vanpelt) [inline wandb/plots/roc.py]: Deps

File: wandb/plots/utils.py [hunk 5]
```diff
@@ -0,0 +1,120 @@
+import wandb
+from wandb import util
+
+# FIXME: look at wandb.util.is_numpy_array() wandb.util.is_pandas* for examples
+np = util.get_module("numpy", required="Logging plots requires numpy")
+pd = util.get_module("pandas", required="Logging dataframes requires pandas")
+scipy = util.get_module("scipy", required="Logging scipy matrices requires scipy")
+spacy = util.get_module("spacy", required="Logging NER and POD requires spacy")
```
Comment (vanpelt) [inline wandb/plots/utils.py]: Again, this will make it so that anyone that imports wandb has these libaries installed regardless of if they use plots.

File: wandb/plots/utils.py [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+sklearn = util.get_module("sklearn")
+collections = util.get_module("collections",
+                    required="Logging python iterables requires collections")
```
Comment (vanpelt) [inline wandb/plots/utils.py]: This is a built in but it's a little tricky.  See: https://github.com/wandb/client/blob/master/wandb/wandb_config.py#L5-L8 You'll want to import Sequence and Iterable directly.

File: wandb/plots/plot_definitions.py (hunk 2 above)
Comment (lavanyashukla) [reply wandb/plots/plot_definitions.py]: yup this is just a reference for how we render the plots in vega.
Vega charts are ephemeral and also the definitions are only available to the creator, so I wanted to save the plot definitions in a more permanent place for anyone to reference.

File: wandb/plots/explain_text.py [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+                        of shape (n_samples, n_classes) with probability values,
+                        i.e. a row per document and a column per output label.
+
+        Returns:
+         Nothing. To see plots, go to your W&B run page.
+
+        Example:
+         wandb.log({'roc': wandb.plots.ExplainText(text, probas)})
+        """
+        eli5 = util.import_module("eli5")
```
[the remark this answers is not in the document]
Comment (lavanyashukla) [reply wandb/plots/explain_text.py]: fixed [here](https://github.com/wandb/client/pull/909/commits/d7aa158d5ee1efa819691dbe0f1c5a8a53f7e152) :)

File: wandb/plots/heatmap.py (hunk 1 above)
Comment (lavanyashukla) [reply wandb/plots/heatmap.py]: fixed [in 909!](https://github.com/wandb/client/pull/909/commits/d7aa158d5ee1efa819691dbe0f1c5a8a53f7e152)

**Comment by lavanyashukla** [comment]
This is ready to merge, except for the [(windows?) test](https://ci.appveyor.com/project/vanpelt/client/builds/31437404) failing that I have no idea how to fix.

File: wandb/plots/utils.py (hunk 6 above)
Comment (lavanyashukla) [reply wandb/plots/utils.py]: fixed [here](https://github.com/wandb/client/pull/909/commits/5095e706af4f7346d0cfdc8d33736d8f32f801f6)

File: wandb/plots/plot_definitions.py (hunk 2 above)
Comment (vanpelt) [reply wandb/plots/plot_definitions.py]: Let's still turn this into markdown.  It will render nicely in GH.

## heatmap/v1

```json
{
    "$schema": "https://vega.github.io/schema/vega-lite/v4.json",
    "padding": 5,
    "width": 500,
    "height": 500,
    "data":
      {
        "name": "${history-table:rows:x-axis,key}"
      },
    "title": {
      "text": {"value": ""}
    },
    "encoding": {
      "x": {
        "field": "x_axis",
        "type": "nominal",
        "axis": { "title": "" }
      },
      "y": {
        "field": "y_axis",
        "type": "nominal",
        "axis": { "title": "" }
      }
    },
    "layer": [
      {
        "mark": "rect",
        "encoding": {
          "color": {
            "field": "values",
            "type": "quantitative",
            "title": "Values",
            "scale": {
              "scheme": "tealblues"
            }
          }
        }
      },
      {
        "mark": "text",
        "encoding": {
          "text": {"field": "values", "type": "quantitative"}
        }
      }
    ]
}
```
